### PR TITLE
Add some e2e release tests for search

### PR DESCRIPTION
**Pull request by ijt**
Test plan: these tests

### CODE REVIEW

File: web/src/e2e/regression.test.ts [hunk 1]
```diff
@@ -195,5 +195,46 @@ describe('regression test suite', () => {
...
+            'Release test 5.10: Perform global, case-sensitive search.',
+            async () => {
+                await driver.page.goto(baseURL + '/search?repo:%5Egithub%5C.com/adjust/go-wrk%24+String+case:yes')
+                await driver.page.waitForFunction(() => document.querySelectorAll('.e2e-search-result').length == 2)
+            },
+            5 * 1000
+        )
+
+        // todo: 5.5 search with repo group
+        // todo: 5.11 filter by fork status and archived status
```
Comment (felixfbecker) [inline web/src/e2e/regression.test.ts]: tip, you can declare these as pending tests by omitting the function parameter

**Comment by ijt** [comment]
> Please edit this file in vscode which has all our warnings/etc built in. I pushed a fix for prettier, but we are now getting more complaints. Please request review once CI is green.

Ah okay, I'll give that a try.

File: web/src/e2e/regression.test.ts (hunk 1 above)
Comment (ijt) [reply web/src/e2e/regression.test.ts]: Done.

**Comment by ijt** [comment]
Checks are passing. @keegancsmith PTAL.

File: web/src/e2e/regression.test.ts [hunk 2]
```diff
@@ -195,5 +195,40 @@ describe('regression test suite', () => {
...
+                await driver.page.goto(baseURL + '/search?q=index:no+^func.*$')
+                await driver.page.waitForFunction(() => document.querySelectorAll('.e2e-search-result').length > 10)
+            },
+            5 * 1000
+        )
+        test(
+            'Release test 5.8: Search for a repository by name.',
+            async () => {
+                await driver.page.goto(baseURL + '/search?q=repo:^auth0/go-jwt-middleware$')
+                await driver.page.waitForFunction(() => document.querySelectorAll('.e2e-search-result').length === 1)
```
Comment (keegancsmith) [inline web/src/e2e/regression.test.ts]: what are you running the regression tests against? Does it setup an external service such that this repo exists as this name and is unique? Or is this hardcoding specifics about a certain setup (like k8s.sgdev)

File: web/src/e2e/regression.test.ts [hunk 3]
```diff
@@ -195,5 +195,40 @@ describe('regression test suite', () => {
...
+            },
+            5 * 1000
+        )
+        test(
+            'Release test 5.10: Perform global, case-sensitive search.',
+            async () => {
+                await driver.page.goto(baseURL + '/search?repo:%5Egithub%5C.com/adjust/go-wrk%24+String+case:yes')
```
Comment (keegancsmith) [inline web/src/e2e/regression.test.ts]: this isn't a global search though? You are specifying a specific repo.

Comment (ijt) [reply web/src/e2e/regression.test.ts]: Good point. Updated the description.

File: web/src/e2e/regression.test.ts (hunk 2 above)
Comment (ijt) [reply web/src/e2e/regression.test.ts]: The `beforeAll` section above sets up the external service with all the repos.

File: web/src/e2e/regression.test.ts [hunk 4]
```diff
@@ -195,5 +195,40 @@ describe('regression test suite', () => {
             },
             5 * 1000
         )
+        test(
+            'Release test 5.1.2: Perform global text search for something with more than 1000 results and use count:1000',
+            async () => {
+                await driver.page.goto(baseURL + '/search?q=.+count:1000')
+                await driver.page.waitForFunction(() => document.querySelectorAll('.e2e-search-result').length > 10)
+            },
+            5 * 1000
```
Comment (felixfbecker) [inline web/src/e2e/regression.test.ts]: FYI, it is better to rely on the timeout of `waitForFunction`/`waitForSelector`/etc (which you can override by passing the `timeout: ` option) than to rely on the Jest timeout. The reason is that one will give you an error like `Timeout waiting for selector ".foo-bar"` with stack trace, while the other will give you an error like `Jest: Test timeout`. For this, the the Jest timeout (here 5s) needs to be smaller than the `waitFor*` timeout (here 30s).

Comment (ijt) [reply web/src/e2e/regression.test.ts]: That does sound better. Do you know why the existing tests don't do that, or is it just that whoever wrote them didn't know?

Comment (ijt) [reply web/src/e2e/regression.test.ts]: I just tried to do it and found that I don't know how. I still think it's a good idea though, so here's an issue to fix them all for consistency: https://github.com/sourcegraph/sourcegraph/issues/5340.

**Pull request merged**